src/transformState.js

```diff
@@ -5,7 +5,27 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const i of actions) {
+    switch (i.type) {
+      case 'addProperties':
+        Object.assign(state, i.extraData);
+        break;
+
+      case 'removeProperties':
+        for (const j of i.keysToRemove) {
+          if (j in state) {
+            delete state[j];
+          }
+        }
+        break;
+
+      case 'clear' :
+        for (const key in state) {
+          delete state[key];
+        }
+        break;
```

Comment:
It would be great to add a default case to avoid potential errors in a situation where the type does not match any of the cases.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/switch#syntax